### ISSUE TITLE
Updating System Settings, don't persist.

### ISSUE THREAD

**Issue by xewl** [AzuraCast/AzuraCast#3525]
**Using Docker installation method**
Yes

**AzuraCast version**
0.11.1 (Stable)

**Host Operating System**
Ubuntu 20.04 LTS

**Describe the bug**
(Afaik: This started happening after I updated from 0.11 to 0.11.1)
After "successfully" changing system settings, the changes don't persist.
This case I wanted to change something in the Security tab;
- Always Use HTTPS -> checked (it is unchecked)
- API "Access-Control-Allow-Origin" header -> * (it's a domain name)

**To Reproduce**
- Go to System Settings,
- Select eg. "Security" tab
- Set `API "Access-Control-Allow-Origin" header` to *
- Save (which it says is successfully)
- Refresh "a few" times

_Result in my case_: it's back to previous value.

**Expected behavior**
Persistence of settings

**Relevant Logs**
Will look into this asap

**Screenshots**
See additional context

**Device(s):**
N/A

**Additional context**
- Sent GIF to @SlvrEagle23 with the case in Discord (because of a domain name being included)
- Message queues are seemingly _empty_

**Comment by SlvrEagle23** [comment]
@xewl I've been unable to reproduce this locally in my testing. Any time I make settings changes, they persist throughout any number of refreshes and are also confirmed to have been saved to the database.

In implementing the more aggressive caching with the new settings entity, I've gone to great lengths to ensure that long-running processes like the workers don't cause trouble, because they reload the settings entity before each new task is processed just to make sure there's never a "stale" version laying around somewhere.

**Comment by xewl** [comment]
I don't really know what to tell you. I of course value your effort.

`docker-compose logs -f` shows me that the POST request does happen, with a subsequent GET.
But the log sadly doesn't really show me anything more than that.

Actually, _anything_ I change in the System Settings; get reset back to what it was after saving. I'm pretty sure the only settings that does _sometimes seems to_ persist at this moment is the "Use WebSockets for Now Playing Updates" (Is it handled differently?) Though, I've seen it reset too.

- The giveaway is that I'm setting both "Instance name", which doesn't change in the sub-header (should happen almost instantly, no?) & "Use WebSockets for Now Playing Updates".
- That "Instance name" field resets, but "Use WebSockets for Now Playing Updates" remains (un)checked.


Is there any way I can provide you the correct information to debug this issue?
According to SysDebugger the message queues are empty (maybe they're faster than I can refresh for)


The only weird thing I can see, but might not be related:
```
web_1                      |
web_1                      | In Connection.php line 374:
web_1                      |
web_1                      |   Could not acknowledge redis message "1607552702876-0".
web_1                      |
web_1                      |
web_1                      | queue:process [--worker-name WORKER-NAME] [--] [<runtime>]
web_1                      |
web_1                      | [2020-12-09T22:25:03.924367+00:00] AzuraCast.NOTICE: Starting new Message Queue worker process. {"runtime":300,"workerName":"app_worker_0"} []
web_1                      |
web_1                      | In Connection.php line 374:
web_1                      |
web_1                      |   Could not acknowledge redis message "1607552704567-0".
web_1                      |
web_1                      |
web_1                      | queue:process [--worker-name WORKER-NAME] [--] [<runtime>]
web_1                      |
web_1                      | [2020-12-09T22:25:05.297405+00:00] AzuraCast.NOTICE: Starting new Message Queue worker process. {"runtime":300,"workerName":"app_worker_0"} []
web_1                      |
web_1                      | In Connection.php line 374:
web_1                      |
web_1                      |   Could not acknowledge redis message "1607552705249-0".
web_1                      |
web_1                      |
web_1                      | queue:process [--worker-name WORKER-NAME] [--] [<runtime>]
web_1                      |
stations_1                 | 2020-12-09 22:25:06,824 INFO exited: tmpreaper (exit status 0; expected)
web_1                      | [2020-12-09T22:25:07.154790+00:00] AzuraCast.NOTICE: Starting new Message Queue worker process. {"runtime":300,"workerName":"app_worker_0"} []
web_1                      | [2020-12-09T22:25:07.484709+00:00] AzuraCast.ERROR: Cannot process media file at path "Patricia Rushen  - Forget Me Nots": ["Cannot find null-byte (0x00) separator between ItemKey #12 and value. ItemKey starts 1413829008 bytes into the APE tag, at file offset 1423696983"] {"file":"/var/azuracast/www/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php","line":80,"code":0} []
web_1                      | [2020-12-09T22:25:07.534443+00:00] AzuraCast.ERROR: Cannot process media file at path "Angela Winbush & Renee  - I'll Be Good": ["Cannot find null-byte (0x00) separator between ItemKey #7 and value. ItemKey starts 1413828929 bytes into the APE tag, at file offset 1419680093"] {"file":"/var/azuracast/www/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php","line":80,"code":0} []
web_1                      | [2020-12-09T22:25:07.591442+00:00] AzuraCast.ERROR: Cannot process media file at path "Johnny Kemp  - Just Got Paid": ["Cannot find null-byte (0x00) separator between ItemKey #8 and value. ItemKey starts 1413828936 bytes into the APE tag, at file offset 1421558684"] {"file":"/var/azuracast/www/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php","line":80,"code":0} []
web_1                      | [2020-12-09T22:25:07.648088+00:00] AzuraCast.ERROR: Cannot process media file at path "Kat Mandu  - I wanna dance": ["Cannot find null-byte (0x00) separator between ItemKey #13 and value. ItemKey starts 1413829058 bytes into the APE tag, at file offset 1419831422"] {"file":"/var/azuracast/www/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php","line":80,"code":0} []
web_1                      | [2020-12-09T22:25:07.695784+00:00] AzuraCast.ERROR: Cannot process media file at path "Kool & The Gang  - Ladies Night": ["Cannot find null-byte (0x00) separator between ItemKey #8 and value. ItemKey starts 1413828939 bytes into the APE tag, at file offset 1418751014"] {"file":"/var/azuracast/www/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php","line":80,"code":0} []
web_1                      | [2020-12-09T22:25:07.754350+00:00] AzuraCast.ERROR: Cannot process media file at path "Shakatak  - City rhythm": ["Cannot find null-byte (0x00) separator between ItemKey #13 and value. ItemKey starts 1413829062 bytes into the APE tag, at file offset 1421152413"] {"file":"/var/azuracast/www/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php","line":80,"code":0} []
web_1                      | [2020-12-09T22:25:07.815568+00:00] AzuraCast.ERROR: Cannot process media file at path "The Jacksons  - Nothin' (That Compares 2 U)": ["Cannot find null-byte (0x00) separator between ItemKey #7 and value. ItemKey starts 1413828934 bytes into the APE tag, at file offset 1421339866"] {"file":"/var/azuracast/www/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php","line":80,"code":0} []
stations_1                 | 2020-12-09 22:25:07,827 INFO spawned: 'tmpreaper' with pid 92674
web_1                      | [2020-12-09T22:25:07.876732+00:00] AzuraCast.ERROR: Cannot process media file at path "Whitney Houston  - Love Will Save The Day": ["Cannot find null-byte (0x00) separator between ItemKey #9 and value. ItemKey starts 1413828968 bytes into the APE tag, at file offset 1421334564"] {"file":"/var/azuracast/www/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php","line":80,"code":0} []
web_1                      | [2020-12-09T22:25:07.926315+00:00] AzuraCast.ERROR: Cannot process media file at path "Cover Girls  - Wishing On A Star": ["Cannot find null-byte (0x00) separator between ItemKey #10 and value. ItemKey starts 1413828999 bytes into the APE tag, at file offset 1420603449"] {"file":"/var/azuracast/www/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php","line":80,"code":0} []
web_1                      | [2020-12-09T22:25:07.984558+00:00] AzuraCast.ERROR: Cannot process media file at path "Michael McDonald & Pattie Label  - On My Own": ["Cannot find null-byte (0x00) separator between ItemKey #11 and value. ItemKey starts 1413829002 bytes into the APE tag, at file offset 1420670669"] {"file":"/var/azuracast/www/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php","line":80,"code":0} []
web_1                      | [2020-12-09T22:25:08.065167+00:00] AzuraCast.ERROR: Cannot process media file at path "Stephanie Mills  - Two Hearts": ["Cannot find null-byte (0x00) separator between ItemKey #13 and value. ItemKey starts 1413829068 bytes into the APE tag, at file offset 1422499941"] {"file":"/var/azuracast/www/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php","line":80,"code":0} []
stations_1                 | 2020-12-09 22:25:08,829 INFO success: tmpreaper entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
```

**Comment by xewl** [comment]
I just updated to Rolling Release @ #63aa00e (2020-12-09 19:27)
The problem explained above, sadly/weirdly persists.

The only thing I've changed in the setup is the following:

```
version: '2.2'

services:
  web:
    volumes:
      - /mnt/volume-mscstrs-1/stations:/var/azuracast/stations
      - /mnt/volume-mscstrs-1/backups:/var/azuracast/backups
      - /mnt/volume-mscstrs-1/media:/var/azuracast/media
      - /mnt/volume-mscstrs-1/recordings:/var/azuracast/recordings

  stations:
    volumes:
      - /mnt/volume-mscstrs-1/stations:/var/azuracast/stations
      - /mnt/volume-mscstrs-1/backups:/var/azuracast/backups
      - /mnt/volume-mscstrs-1/media:/var/azuracast/media
      - /mnt/volume-mscstrs-1/recordings:/var/azuracast/recordings
      - /mnt/volume-mscstrs-1/setup/error.mp3:/usr/local/share/icecast/web/error.mp3
      - /mnt/volume-mscstrs-1/setup/intro:/usr/local/share/icecast/web/intro
```

So, yeah, those files in question are on a separate virtual volume, for exactly these purposes.
All files seems to have the correct rights though.. [intro is currently not used, yet]
Not sure if anything "System Settings" related would be saved here? [Given they go in the DB?]

**Comment by SlvrEagle23** [comment]
@xewl You actually may have inadvertently pointed out another issue (the Redis exception) that led me down a separate rabbit hole and led to me fixing a long-standing issue with file processing that was causing files to process at an extraordinarily slow rate, which is now thankfully resolved as well.

I also did a bit of tinkering to the new settings storage system which should entirely eliminate any issue of settings being overwritten by long-running tasks.

**Comment by xewl** [comment]
This issue is now resolved since I updated [#6671882] - Someone finding this, please update to at least 0.11.2

Thank you @SlvrEagle23 !
